File: VB-Examples/02_Data/MergeScenario.vb
Imports Spire.Xls
Imports System
Imports System.Windows.Forms

Namespace MergeScenario
    Public Partial Class Form1 : Inherits Form
        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs)
            ' Create a workbook
            Dim wb As Workbook = New Workbook()
            Dim inputFile As String = @"..\..\..\..\..\..\Data\ScenarioSample4.xlsx"
            wb.LoadFromFile(inputFile)
            Dim worksheet1 As Worksheet = wb.Worksheets[0]
            Dim worksheet2 As Worksheet = wb.Worksheets[1]

            'Merge the scenario 
            worksheet1.Scenarios.Merge(worksheet2)

            'Saving the workbook
            Dim outputFile As String = "MergeScenario.xlsx"
            wb.SaveToFile(outputFile, ExcelVersion.Version2013)
            wb.Dispose()

            'View the document
            FileViewer(outputFile)
        End Sub

        Private Sub FileViewer(ByVal fileName As String)
            Try
                System.Diagnostics.Process.Start(fileName)
        Catch
        End Try
        End Sub

        Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs)
            Close()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)

        End Sub
    End Class
End Namespace

File: CS-Examples/02_Data/MergeScenario.cs
using Spire.Xls;
using System;
using System.Windows.Forms;

namespace MergeScenario
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            // Create a workbook
            Workbook wb = new Workbook();
            String inputFile = @"..\..\..\..\..\..\Data\ScenarioSample4.xlsx";
            wb.LoadFromFile(inputFile);
            Worksheet worksheet1 = wb.Worksheets[0];
            Worksheet worksheet2 = wb.Worksheets[1];

            //Merge the scenario 
            worksheet1.Scenarios.Merge(worksheet2);

            //Saving the workbook
            String outputFile = "MergeScenario.xlsx";
            wb.SaveToFile(outputFile, ExcelVersion.Version2013);
            wb.Dispose();

            //View the document
            FileViewer(outputFile);
        }

        private void FileViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }

        private void Form1_Load(object sender, EventArgs e)
        {

        }
    }
}
